mk/common-test.sh

```diff
@@ -1,15 +1,27 @@
-test_dir=tests/functional
+# Remove overall test dir (at most one of the two should match) and
+# remove file extension.
+test_name=$(echo -n "$test" | sed \
+    -e "s|^unit-test-data/||" \
+    -e "s|^tests/functional/||" \
+    -e "s|\..*$||" \
```

Comment:
```suggestion
    -e "s|\.[a-z]*$||" \
```

I don't think we use periods on the name, but this will make it less greedy for something like `fetchers-github.com.sh` or whatever. But do we really need to strip arbitrary extensions in the first place?

```suggestion
    -e "s|\.sh$||" \
```